### PR TITLE
Create 105B/E feature flag and skeleton #162620473

### PR DESCRIPTION
**Pull request by donaldthai**
## Description

This PR creates the feature flag for the "More Robust Accessorial" and a skeleton component of 105B/E accessorial item. This will enable the pre-approval section to support custom inputs upon selecting the `Code & Item` input.

## Reviewer Notes

- Currently setting the feature flag to be disabled for Staging and Production.

## Code Review Verification Steps

* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/162620473) for this change

## Screenshots

jan-15-2019 16-16-15

### CODE REVIEW

File: src/shared/PreApprovalRequest/DetailsHelper.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { DefaultDetails } from './DefaultDetails';
+import { Code105Details } from './Code105Details';
+
+export function getDetailComponent(code, robustAccessorial) {
```
Comment (tinyels) [inline src/shared/PreApprovalRequest/DetailsHelper.js]: Please write a jest test around this function.

Comment (donaldthai) [reply src/shared/PreApprovalRequest/DetailsHelper.js]: 👍

Comment (donaldthai) [reply src/shared/PreApprovalRequest/DetailsHelper.js]: @tinyels unit test added!

File: src/shared/PreApprovalRequest/DetailsHelper.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { getDetailComponent } from './DetailsHelper';
+
+let wrapper;
+describe('testing getDetailComponent()', () => {
+  describe('returns default details component', () => {
```
Comment (tinyels) [inline src/shared/PreApprovalRequest/DetailsHelper.test.js]: I think we want tests here that confirm 2 things:

- The feature flag is working
- We only get the 105 component when the code is 105
- in all other cases we get the default component

We don't need to render to perform these tests.

File: src/shared/PreApprovalRequest/Code105Details.test.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import { shallow } from 'enzyme';
+import { Code105Details } from './Code105Details';
+
+let wrapper;
+describe('Renders without crashing', () => {
+  describe('Code 105B/E details component', () => {
+    wrapper = shallow(<Code105Details />);
+
+    it('renders without crashing', () => {
+      // eslint-disable-next-line
```
Comment (lynzt) [inline src/shared/PreApprovalRequest/Code105Details.test.js]: why do we need to do this? I'm just curious...

File: src/shared/PreApprovalRequest/PreApprovalForm.jsx [hunk 4]
```diff
@@ -157,9 +140,10 @@ PreApprovalForm = reduxForm({
...
 function mapStateToProps(state) {
   return {
+    tariff400ng_item_code: get(state, 'form.preapproval_request_form.values.tariff400ng_item.code'),
     ship_line_item_schema: get(state, 'swaggerPublic.spec.definitions.ShipmentLineItem', {}),
     filteredLocations: selectLocationFromTariff400ngItem(state),
   };
 }
 
-export default connect(mapStateToProps)(PreApprovalForm);
+export default withContext(connect(mapStateToProps)(PreApprovalForm));
```
Comment (lynzt) [inline src/shared/PreApprovalRequest/PreApprovalForm.jsx]: curious what `withContext` does...

Comment (donaldthai) [reply src/shared/PreApprovalRequest/PreApprovalForm.jsx]: @tinyels can probably answer this better but from looking in `shared/AppContext/index.jsx` file `withContext` creates [React context](https://reactjs.org/docs/context.html) which we make available to the Redux store. In this case, we're using this to be able to call the feature flag constants: `robustAccessorial`

File: src/shared/PreApprovalRequest/Code105Details.test.js (hunk 3 above)
Comment (donaldthai) [reply src/shared/PreApprovalRequest/Code105Details.test.js]: oh this is to disable the warnings from eslint :)

Comment (lynzt) [reply src/shared/PreApprovalRequest/Code105Details.test.js]: what is the warning? what does eslint not like about the line? (I guess I could run it and check myself... )

Comment (donaldthai) [reply src/shared/PreApprovalRequest/Code105Details.test.js]: oh, just double checked and there weren't any es lint warnings when I removed them! I did templated the tests off of existing tests.

File: src/shared/PreApprovalRequest/DetailsHelper.test.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+    it('renders without crashing', () => {
+      expect(DetailComponent).toBe(DefaultDetails);
+    });
+  });
+
+  describe('returns default details component with feature flag off', () => {
+    //pass in known code item with feature flag off
+    let DetailComponent = getDetailComponent('105', false);
+    it('renders default details without crashing', () => {
```
Comment (tinyels) [inline src/shared/PreApprovalRequest/DetailsHelper.test.js]: this comment doesn't match what the test is doing.

File: src/shared/PreApprovalRequest/DetailsHelper.test.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      expect(DetailComponent).toBe(DefaultDetails);
+    });
+  });
+
+  describe('returns 105B/E details component with feature flag on', () => {
+    let DetailComponent = getDetailComponent('105', true);
+    it('renders 105 details without crashing', () => {
```
Comment (tinyels) [inline src/shared/PreApprovalRequest/DetailsHelper.test.js]: It's not that it renders; it returns a 105 details component.

File: src/shared/PreApprovalRequest/DetailsHelper.test.js (hunk 2 above)
Comment (tinyels) [reply src/shared/PreApprovalRequest/DetailsHelper.test.js]: It's better, but I think we want at least 2 tests that cover one of the other codes.

**Pull request merged**